src/hotspot/share/opto/multnode.hpp

```diff
@@ -105,4 +106,30 @@ class ProjNode : public Node {
   ProjNode* other_if_proj() const;
 };
 
+//------------------------------TupleNode---------------------------------------
+class TupleNode : public MultiNode {
+  const TypeTuple* _tf;
+
+  template <typename... NN>
```

Comment:
Does this need to be a template? Or would a type like `Node*` or `Node` suffice?

Reply:
The point is that it's a variadic template. Of course, ideally, I'd like it to be a pack of `Node*` but there isn't a simple way to write that. The upside is that one can write `TupleNode::make(some_type, input1, input2, input3, input4)` with how many input you want in a single construction. I prefer hiding the not nice here, to have compact and readable usages of TupleNode, rather than having every usage look like
```cpp
TupleNode* tuple = new TupleNode(some_type);
tuple.set_req(0, input1);
tuple.set_req(1, input2);
tuple.set_req(2, input3);
tuple.set_req(3, input4);
```

Reply:
Yes, it looks good enough :)